rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -219,21 +223,23 @@ void RSocketStateMachine::closeFrameTransport(
   }
 
   if (resumeCallback_) {
-    resumeCallback_->onResumeError(ConnectionException(
+    auto resumeCallback = std::move(resumeCallback_);
+    resumeCallback->onResumeError(ConnectionException(
         ex ? ex.get_exception()->what() : "connection closing"));
-    resumeCallback_.reset();
   }
 
   // Echo the exception to the frameTransport only if the frameTransport started
   // closing with error.  Otherwise we sent some error frame over the wire and
   // we are closing the transport cleanly.
-  if (signal == StreamCompletionSignal::CONNECTION_ERROR) {
-    frameTransport_->closeWithError(std::move(ex));
-  } else {
-    frameTransport_->close();
+  if (frameTransport_) {
```

Comment:
The bug fix here:

In the same situation as above, after onResumeError() gets invoked, RSocketClient can be destroyed which can lead to frameTransport_ being reset.  And we will be accessing a nullptr here.